Sources/HTTP/HTTPServer.swift

```diff
@@ -26,15 +26,33 @@ public protocol HTTPServing : class {
 /// abstraction, but the intention is to remove this dependency and reimplement
 /// the class using transport APIs provided by the Server APIs working group.
 public class HTTPServer: HTTPServing {
+
+    /// A subclass for HTTP Options
+    /// HTTPServer to be created on the given `port`, using `handler` to process incoming requests
+    public class Options {
```

Comment:
Can this be a `struct`?

Reply:
Good catch, it should be a struct if the class is `public` anyways. It should be `open` instead. (Same maybe for `HTTPServer`.)

Also: the comment is wrong `/// A subclass for HTTP Options`. This is not a subclass but a root class.

Struct vs Class: Depends a little on whether we want to allow subclassing. The idea to use a class comes from my 2017-10-30 email:

    class HTTPServer {
     class Options { // a class because subclasses may want to add

It could be a struct, but this would require wrapping if you add fields to it (very common in JS style `options` arrays). E.g. a framework on top, which wraps HTTPServer, could still use the same configuration, but add a few things (say a logger, or some template path).

Personally I don't care to much in this case. Either would be fine for me. But if it is a class, it needs to be `open`, that is the whole point.

Reply:
👍

Reply:
I like the idea of `open class` for options (for example, it can be used by `PoCSocketSimpleServer`) but then should `Options` be defined within `HTTPServer`? (we do get namespacing, but will it be cleaner if `Options` is defined outside of `HTTPServer`?

Also what use is `open` Options but only `public` HTTPServer?

Good pick up on the comment. Thanks!

Reply:
> Also what use is open Options but only public HTTPServer.

```swift
public class MyExpressServer {
  open class Options : HTTPServer.Options {
    let templatePath : String
  }
  let http : HTTPServer
}
```

But maybe it should be just a struct. Seems cleaner for such an isolated project.

P.S.: We could have a general discussion of whether we want the classes `open` or closed. I'm generally in favour of `open`, because you never know, but Swift style may be different.

Reply:
Your example is exactly what I wanted to use for `PoCSockerSimpleServer` that I mentioned.
And what sold me on making options open.  :-P